tests/contracts/wormhole-token-bridge/src/spec.rs

```diff
@@ -0,0 +1,391 @@
+use std::{
+    convert::{TryFrom, TryInto},
+    io::{Cursor, Read},
+    str,
+};
+
+use byteorder::{BigEndian, ReadBytesExt};
+use sha3::{Digest, Keccak256};
+
+use oasis_wormhole_types as wormhole;
+
+use crate::Error;
+
+/// Computes internal wrapped asset ID from the assets chain ID and it's address.
+/// XXX: This is how terra computes the wrapped asset id's, but we could use a different scheme.
```

Comment:
Are these fixed-size? Why does it need to be hashed? Is this only used internally and not in any protocol messages?

Reply:
Only used internally. Yeah no need for hashing, we can just concat the id and address as both are fixed sized